lib/arrays.md

```diff
@@ -5,44 +5,105 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
+giving some kind of award to the first 20 finishers. Concert ticket orders that are being filled on a first-come, first-served basis.
+
 
 # What is one way, using Ruby, to retrieve the 6th element in an Array? How about the 8th element? What happens if you try to retrieve the value of the _9999th_ element (or some element that doesn't exist in the array)?
 
-Replace me with your response...
+You can ask Ruby for the 6th element. Say I have an array called TestArray. Because I like to see output, I would write puts TestArray[6] to 
+show the element in slot 6. For the 8th element, change TestArray[6] to TestArray[8]. 
 
 # The previous question asks about finding, for example, the 6th element in an Array. Is it possible to find the **-6th** (Notice the negative symbol!) element in an Array? What does that even mean?
 
-Replace me with your response...
+The code line "puts TestArray[-6]" would produce the 6th item from the end of the array. 
 
 # How would you perform an operation on every element inside an Array?
 
-Replace me with your response...
+To perform an operation on every element in an array, or even a range of them, you would use a loop.
+In repl.it I made an array with the names of the seven dwarves in the Snow White story.
+(First I put "Grouchy" or "Grumpy" but fixed it.) Anyway, the code below produces the names in the array one at a time.
+From The Bastard's Book it looks like there would be several ways to do make a loop that does this.
+
+TestArray.each do |x|
+    puts x
+end
 
 # How do you add elements to an Array?
 
-Replace me with your response...
+With Ruby there seem to be several ways to do anything. In repl.it I used the "push" method. It adds new items after the last item in the 
+existing array. The program below returns a list of 10 Snow White characters beginning with "Happy" and ending with "the Evil Queen."
+Note to me: At least on repl.it, and I extrapolate from that everywhere, if I have a space, as in TestArray.push (variables) 
+instead of TestArray.push(variables) I get an error message. 
+
+TestArray = ["Happy", "Dopey", "Sleepy", "Sneezy", "Grumpy", "Bashful", "Doc"]
+TestArray.push("Snow White", "the Huntsman", "the Evil Queen")
+puts TestArray.join(", ")
+
 
 # Given the Array `["Laura", "Fiona", "Tori"]`, how would you replace `"Fiona"` with `"Florence"` so that you end up with `["Laura", "Florence", "Tori"]`?
 
-Replace me with your response...
+This "works" but there must be a better way to do it. It returned the "["Laura", "Florence," Tori"]" array but seems limited. If you had 
+25 replacements instead of one making a separate array for each one would be clunky. I learned a little about "map" method, and not-enough-yet about what a block is. 
+#Replace Fiona with Florence in an array
+Names = ["Laura", "Fiona", "Tori"]
+NewName = ["Florence"]
+Names.map! do |x| x=="Fiona"? NewName: x 
```

Comment:
You're right that there is a better way. Consider a standard variable `x` which we have previously defined as `"Sumeet"`.

To get the value of `x`, we would write the following code: `x`. Simple! :) How about to change the value of `x`? We would write `x = "Patrick"`–that is, we would use the "assignment operator".

It's just about as easy with array elements. To get an array element's value, you simply refer to it: `Names[0]` would return `"Laura"`. To **set** an array element's value, you could use the assignment operator here as well. I'll let you take it from here :)